obsreport/obsreport_processor.go

```diff
@@ -54,134 +84,218 @@ type ProcessorSettings struct {
 
 // NewProcessor creates a new Processor.
 func NewProcessor(cfg ProcessorSettings) (*Processor, error) {
-	return &Processor{
-		level:    cfg.ProcessorCreateSettings.MetricsLevel,
-		mutators: []tag.Mutator{tag.Upsert(obsmetrics.TagKeyProcessor, cfg.ProcessorID.String(), tag.WithTTL(tag.TTLNoPropagation))},
-	}, nil
+	return newProcessor(cfg, featuregate.GetRegistry())
+}
+
+func newProcessor(cfg ProcessorSettings, registry *featuregate.Registry) (*Processor, error) {
+	proc := &Processor{
+		level:             cfg.ProcessorCreateSettings.MetricsLevel,
+		mutators:          []tag.Mutator{tag.Upsert(obsmetrics.TagKeyProcessor, cfg.ProcessorID.String(), tag.WithTTL(tag.TTLNoPropagation))},
+		logger:            cfg.ProcessorCreateSettings.Logger,
+		useOtelForMetrics: registry.IsEnabled(obsreportconfig.UseOtelForInternalMetricsfeatureGateID),
+		meter:             cfg.ProcessorCreateSettings.MeterProvider.Meter(processorScope),
+		otelAttrs: []attribute.KeyValue{
+			attribute.String(obsmetrics.ProcessorKey, cfg.ProcessorID.String()),
+		},
+	}
+
+	if err := proc.createOtelMetrics(); err != nil {
+		return nil, err
+	}
+
+	return proc, nil
+}
+
+func (por *Processor) createOtelMetrics() error {
```

Comment:
Let's keep the same pattern of passing the configs instance as parameter here and creating the meter inside this function as [the previous prs](https://github.com/open-telemetry/opentelemetry-collector/pull/6460/files#diff-2864c8597e07e090876715c3acba587547f0f86254491f01cc00faa7be1c301fR107)

Reply:
Thanks for the review! Fixed!